internal/pkg/bootstrap/handlers/secret/secret.go

```diff
@@ -63,3 +72,34 @@ func BootstrapHandler(
 
 	return err == nil
 }
+
+// getSecretConfig creates a SecretConfig based on the SecretStoreInfo configuration properties.
+// If a tokenfile is present it will override the Authentication.AuthToken value.
+func getSecretConfig(secretStoreInfo config.SecretStoreInfo) (vault.SecretConfig, error) {
```

Comment:
I am not completely a fan of doing this logic at this level. I think that it would be best if this lived in `go-mod-secrets` and the `SecretConfig` strruct containing the `tokenfile` field. That way consumers of the `go-mod-secrets` module can simply pass the configuration properties to the constructor without having to perform this logic. The reason this cannot be done now is that the logic which parses the file lives in `edgex-go` and moving the logic to `go-mod-secrets` would create a cyclic dependency. If this seems reasonable I will create a followup issue to migrate the `AuthTokenLoader` functionality to `go-mod-secrets`. I would like to not tackle this effort here as we are approaching a code freeze for the Fuji release.

Reply:
> I am not completely a fan of doing this logic at this level.

Agreed.  Create a follow up issue for post-Fuji resolution to move this code into go-mod-secrets.